tests/account.write.test.ts

```diff
@@ -118,110 +128,103 @@ describe("account.write", () => {
-  //       MOCK_FALLBACK_HANDLER,
-  //       ModuleType.Fallback,
-  //       deInitData
-  //     )
-
-  //     const isInstalled = await smartAccount.isModuleInstalled(
-  //       ModuleType.Fallback,
-  //       MOCK_FALLBACK_HANDLER,
-  //       ethers.AbiCoder.defaultAbiCoder().encode(
-  //         ["bytes4"],
-  //         [GENERIC_FALLBACK_SELECTOR as Hex]
-  //       ) as Hex
-  //     )
-
-  //     expect(userOpReceipt.success).toBe(true)
-  //     expect(isInstalled).toBeFalsy()
-  //     expect(userOpReceipt).toBeTruthy()
-  //   }, 60000)
+    test("install a mock Hook module", async () => {
+      const isSupported = await smartAccount.supportsModule("hook")
+      console.log(isSupported, "is supported")
+
+      const isInstalledBefore = await smartAccount.isModuleInstalled({
+        moduleAddress: MOCK_HOOK,
+        type: "hook"
+      })
+
+      console.log(isInstalledBefore, "is installed before")
+
+      const { wait } = await smartAccount.installModule({
+        moduleAddress: MOCK_HOOK,
+        type: "hook"
+      })
+      const { success } = await wait()
+     
+      const isInstalled = await smartAccount.isModuleInstalled({
+        moduleAddress: MOCK_HOOK,
+        type: "hook"
+      })
+      expect(success).toBe(true)
+      expect(isInstalled).toBeTruthy()
+    }, 60000)
+
+    test("get active hook", async () => {
+      const activeHook: Address = await smartAccount.getActiveHook()
+      console.log(activeHook, "active hook")
+      expect(activeHook).toBe(MOCK_HOOK)
+    }, 60000)
+
+    test("uninstall hook module", async () => {
+      const prevAddress: Hex = "0x0000000000000000000000000000000000000001"
+      const deInitData = encodeAbiParameters(
+        [
+          { name: "prev", type: "address" },
+          { name: "disableModuleData", type: "bytes" }
+        ],
+        [prevAddress, toHex(stringToBytes(""))]
+      )
+      const { wait }  = await smartAccount.uninstallModule(
+        {
+          moduleAddress: MOCK_HOOK,
+          type: "hook",
+          data: deInitData // review
+        }
+      )
+
+      const { success } = await wait()
+
+      const isInstalled = await smartAccount.isModuleInstalled({
+        moduleAddress: MOCK_HOOK,
+        type: "hook"
+      })
+
+      expect(success).toBe(true)
+      expect(isInstalled).toBeFalsy()
+    }, 60000)
+
+    test("install a fallback handler Hook module", async () => {
+      const isSupported = await smartAccount.supportsModule("fallback")
+      console.log(isSupported, "is supported")
+
+      const isInstalledBefore = await smartAccount.isModuleInstalled({
+        type: "fallback",
+        moduleAddress: MOCK_FALLBACK_HANDLER,
+        data: encodeAbiParameters(
+          [{ type: 'bytes4' }],
+          [ GENERIC_FALLBACK_SELECTOR_SELECTOR as Hex ]
+        )
+      })
+      console.log(isInstalledBefore, "is installed before")
+    }, 60000)
+
+    // test("uninstall handler module", async () => {
```

Comment:
these can't run until installModule and uinstallModule is properly written in Nexus. along with accepted types reviews